contributingGuides/PROPTYPES_CONVERSION_TABLE.md

```diff
@@ -0,0 +1,142 @@
+| `PropTypes.instanceOf(T)`                                            | `T`                                           | Convert to `T`.                                                                                                                                                                                                                               |
+| `PropTypes.oneOf([T, U, ...])` or `PropTypes.oneOfType([T, U, ...])` | `T \| U \| ...`                               | Convert to a union type e.g. `T \| U \| ...`.                                                                                                                                                                                                 |
+
+## Conversion Example
+
+```ts
+// Before
+const propTypes = {
+  unknownData: PropTypes.any,
+  anotherUnknownData: PropTypes.any,
+  indexes: PropTypes.arrayOf(PropTypes.number),
+  items: PropTypes.arrayOf(
+    PropTypes.shape({
+      value: PropTypes.string,
+      label: PropTypes.string,
+    })
+  ),
+  shouldShowIcon: PropTypes.bool,
+  onChangeText: PropTypes.func,
+  count: PropTypes.number,
+  session: PropTypes.shape({
+    authToken: PropTypes.string,
+    accountID: PropTypes.number,
+  }),
+  errors: PropTypes.objectOf(PropTypes.string),
+  inputs: PropTypes.objectOf(
+    PropTypes.shape({
+      id: PropTypes.string,
+      label: PropTypes.string,
+    })
+  ),
+  label: PropTypes.string,
+  anchor: PropTypes.node,
+  footer: PropTypes.element,
+  uniqSymbol: PropTypes.symbol,
+  icon: PropTypes.elementType,
+  date: PropTypes.instanceOf(Date),
+  size: PropTypes.oneOf(["small", "medium", "large"]),
+
+  optionalString: PropTypes.string,
+  /**
+   * Note that all props listed above are technically optional because they lack the `isRequired` attribute.
+   * However, in most cases, props are actually required but the `isRequired` attribute is left out by mistake.
+   *
+   * For each prop that appears to be optional, determine whether the component implementation assumes that
+   * the prop has a value (making it non-optional) or not. Only those props that are truly optional should be
+   * labeled with a `?` in their type definition.
+   */
+};
+
+// After
+type Item = {
+  value: string;
+  label: string;
+};
+
+type Session = {
+  authToken: string;
+  accountID: number;
+};
+
+type Input = {
+  id: string;
+  label: string;
+};
+
+type Size = "small" | "medium" | "large";
+
+type Props = {
+  unknownData: string[];
+
+  // It's not possible to infer the data as it can be anything because of reasons X, Y and Z.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  anotherUnknownData: any;
```

Comment:
```suggestion
  // It's not possible to infer the data as it can be anything because of reasons X, Y and Z.
  anotherUnknownData: unknown;
```

@hayata-suenaga Let's not encourage usage of `any` here.

Reply:
nice catch 👍